### ISSUE TITLE
Return "invalid option: XXX" when use -R option

### ISSUE THREAD

**Issue by mikoto2000** [embulk/embulk#290]
## Problem

Return "invalid option: XXX" when use -R option

```sh
$ ../embulk/pkg/embulk-0.7.3.jar -R-U migrate ./
2015-08-29 10:46:54.442 +0900: Embulk v0.7.3
Usage: migrate <directory>

invalid option: -U
```

## Cause

Cause is order of "org.jruby.Main.main" method's arguments.

### "org.jruby.Main.main" method's arguments of Embulk v0.7.3

1. path of 'embulk_bundle.rb'
2. jruby args
3. subcommand
4. subcommand args

Jruby args position have to set before "path of 'embulk_bundle.rb'".

## Draft of fix

Have to fix jruby args position, before "path of 'embulk_bundle.rb'".

1. jruby args
2. path of 'embulk_bundle.rb'
3. subcommand
4. subcommand args

### Fix of selfrun.sh

Add "EMBULK_BUNDLE_RB" between "jruby_args" and "args".

[embulk/selfrun.sh at master · embulk/embulk #L41](https://github.com/embulk/embulk/blob/master/embulk-cli/src/main/sh/selfrun.sh#L41)

[embulk/selfrun.sh at master · embulk/embulk #L157](https://github.com/embulk/embulk/blob/master/embulk-cli/src/main/sh/selfrun.sh#L157)

### Fix of 'org.embulk.cli.Main.main'

Retrieval args and replace '"EMBULK_BUNDLE_RB"' to 'resourcePath + "embulk/command/embulk_bundle.rb"'.

[embulk/Main.java at master · embulk/embulk](https://github.com/embulk/embulk/blob/master/embulk-cli/src/main/java/org/embulk/cli/Main.java#L18)

Is there any problem?

**Comment by mikoto2000** [comment]
This draft have problem.
If use java command, need input EMBULK_BUNDLE_RB string.

## Example

```sh
$ java -jar embulk-0.7.3.jar -U-EUTF-8 EMBULK_BUNDLE_RB migrate ./
```


## Draft of fix 2.

### Fix of 'org.embulk.cli.Main'

1. Add list of subcommand.
2. Search subcommand in args[].
3. Insert 'resourcePath + "embulk/command/embulk_bundle.rb"', before subcommand.

May I create pull request of draft2?

**Comment by frsyuki** [comment]
No, because problem is in `embulk-cli/src/main/sh/selfrun.sh`. `org.embulk.cli.Main` is the component that should handle JRuby options because it starts `org.jruby.Main.main`. Therefore selfrun.sh shouldn't be aware of -R option.
Fixed at 813dde1.
